pkg/plugins/external/helpers.go

```diff
@@ -0,0 +1,118 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package external
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+
+	"sigs.k8s.io/kubebuilder/v3/pkg/machinery"
+	"sigs.k8s.io/kubebuilder/v3/pkg/plugin/external"
+)
+
+var outputGetter ExecOutputGetter = &execOutputGetter{}
+
+// ExecOutputGetter is an interface that implements the exec output method.
+type ExecOutputGetter interface {
+	GetExecOutput(req []byte, path string) ([]byte, error)
+}
+
+type execOutputGetter struct{}
+
+func (e *execOutputGetter) GetExecOutput(request []byte, path string) ([]byte, error) {
+	cmd := exec.Command(path) //nolint:gosec
+	cmd.Stdin = bytes.NewBuffer(request)
+	cmd.Stderr = os.Stderr
+	out, err := cmd.Output()
+	if err != nil {
+		return nil, err
+	}
+
+	return out, nil
+}
+
+var currentDirGetter OsWdGetter = &osWdGetter{}
+
+// OsWdGetter is an interface that implements the get current directory method.
+type OsWdGetter interface {
+	GetCurrentDir() (string, error)
+}
+
+type osWdGetter struct{}
+
+func (o *osWdGetter) GetCurrentDir() (string, error) {
+	currentDir, err := os.Getwd()
+	if err != nil {
+		return "", fmt.Errorf("error getting current directory: %v", err)
+	}
+
+	return currentDir, nil
+}
+
+func getPluginResponse(fs machinery.Filesystem, req external.PluginRequest, path string) error {
+	req.Universe = map[string]string{}
+
+	reqBytes, err := json.Marshal(req)
+	if err != nil {
+		return err
+	}
+
+	out, err := outputGetter.GetExecOutput(reqBytes, path)
+	if err != nil {
+		return err
+	}
+
+	res := external.PluginResponse{}
+	if err := json.Unmarshal(out, &res); err != nil {
+		return err
+	}
+
+	// Error if the plugin failed.
+	if res.Error {
+		return fmt.Errorf(strings.Join(res.ErrorMsgs, "\n"))
+	}
+
+	currentDir, err := currentDirGetter.GetCurrentDir()
+	if err != nil {
+		return fmt.Errorf("error getting current directory: %v", err)
+	}
```

Comment:
Just pass the current wd as an argument to this function.

Reply:
@estroz do you mean that I could just parameterize `GetCurrentDir` by passing current wd as an argument, and in the unit test I could just mock that value, instead of creating an [interface](https://github.com/kubernetes-sigs/kubebuilder/pull/2338/files#diff-d40306bd214f2bcd22890c26bd1d40a49b0dfba323b99814c8fcf93d85543d0dR56) for it?

Reply:
Yep!